File: Practico3/Form1.vb
Public Class Form1
    Private Sub LNya_Click(sender As Object, e As EventArgs) Handles LNya.Click

    End Sub

    Private Sub TDni_TextChanged(sender As Object, e As EventArgs) Handles TDni.TextChanged
        Dim input As String = TDni.Text

        ' Verificar si el texto contiene caracteres que no sean números
        For Each c As Char In input
            If Not Char.IsDigit(c) Then
                ' Si se encuentra un carácter no numérico, eliminarlo del TextBox
                TDni.Text = TDni.Text.Remove(TDni.Text.IndexOf(c), 1)
                TDni.SelectionStart = TDni.Text.Length ' Colocar el cursor al final del TextBox
                Exit For
            End If
        Next
    End Sub

    Private Sub TApellido_TextChanged(sender As Object, e As EventArgs) Handles TApellido.TextChanged
        Dim input As String = TApellido.Text

        ' Verificar si el texto contiene caracteres que no sean letras
        For Each c As Char In input
            If Not Char.IsLetter(c) Then
                ' Si se encuentra un carácter no alfabético, eliminarlo del TextBox
                TApellido.Text = TApellido.Text.Remove(TApellido.Text.IndexOf(c), 1)
                TApellido.SelectionStart = TApellido.Text.Length ' Colocar el cursor al final del TextBox
                Exit For
            End If
        Next
    End Sub

    Private Sub TNombre_TextChanged(sender As Object, e As EventArgs) Handles TNombre.TextChanged
        Dim input As String = TNombre.Text

        ' Verificar si el texto contiene caracteres que no sean letras
        For Each c As Char In input
            If Not Char.IsLetter(c) Then
                ' Si se encuentra un carácter no alfabético, eliminarlo del TextBox
                TNombre.Text = TNombre.Text.Remove(TNombre.Text.IndexOf(c), 1)
                TNombre.SelectionStart = TNombre.Text.Length ' Colocar el cursor al final del TextBox
                Exit For
            End If
        Next
    End Sub

    Private Sub TGuardar_Click(sender As Object, e As EventArgs) Handles TGuardar.Click
        Dim ask As MsgBoxResult ' Declarar la variable ask

        If String.IsNullOrWhiteSpace(TDni.Text) OrElse
       String.IsNullOrWhiteSpace(TNombre.Text) OrElse
       String.IsNullOrWhiteSpace(TApellido.Text) Then

            ' Mostrar mensaje de error si algún campo está incompleto
            MsgBox("Debe Completar todos los campos", MsgBoxStyle.Critical, "Error")
        Else
            ' Concatenar apellido y nombre y asignar a LModificar.Text
            LModificar.Text = TNombre.Text & " " & TApellido.Text

            ' Mostrar mensaje de confirmación
            ask = MsgBox("Seguro que desea insertar un nuevo Cliente?", MsgBoxStyle.YesNo + MsgBoxStyle.DefaultButton1, "Confirmar Insercion")
            If ask = MsgBoxResult.Yes Then

                MsgBox("Cliente insertado exitosamente", MsgBoxStyle.Information, "Éxito")
            End If
        End If
    End Sub

    Private Sub TEliminar_Click(sender As Object, e As EventArgs) Handles TEliminar.Click
        Dim ask As MsgBoxResult ' Declarar la variable ask

        ' Mostrar mensaje de advertencia para confirmar eliminación
        ask = MsgBox("Está a punto de eliminar el Cliente: " & TNombre.Text & " " & TApellido.Text, MsgBoxStyle.YesNo + MsgBoxStyle.DefaultButton2 + MsgBoxStyle.Exclamation, "Confirmar Eliminación")

        If ask = MsgBoxResult.Yes Then
            ' Limpiar los campos
            TDni.Clear()
            TNombre.Clear()
            TApellido.Clear()
            LModificar.Text = "Modificar"

            MsgBox("Cliente eliminado exitosamente", MsgBoxStyle.Information, "Éxito")
        End If
    End Sub

    Private Sub RBVaron_CheckedChanged(sender As Object, e As EventArgs) Handles RBVaron.CheckedChanged

        If RBVaron.Checked Then
            PictureBox1.BackgroundImage = My.Resources.varonIcon
        End If
    End Sub

    Private Sub BSalir_Click(sender As Object, e As EventArgs) Handles BSalir.Click
        Me.Close()
    End Sub

    Private Sub PictureBox1_Click(sender As Object, e As EventArgs) Handles PictureBox1.Click

    End Sub

    Private Sub RBMujer_CheckedChanged(sender As Object, e As EventArgs) Handles RBMujer.CheckedChanged
        If RBVaron.Checked Then
            PictureBox1.BackgroundImage = My.Resources.mujerIcon
        End If
    End Sub
End Class

File: Practico3/Form1.cs
namespace Practico3
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void TDni_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsDigit(e.KeyChar) && !char.IsControl(e.KeyChar))
            {
                e.Handled = true;
            }
        }

        // Validar que solo se ingresen letras en TApellido y TNombre
        private void TNombre_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsLetter(e.KeyChar) && !char.IsControl(e.KeyChar) && !char.IsWhiteSpace(e.KeyChar))
            {
                e.Handled = true;
            }
        }

        private void TApellido_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsLetter(e.KeyChar) && !char.IsControl(e.KeyChar) && !char.IsWhiteSpace(e.KeyChar))
            {
                e.Handled = true;
            }
        }

        private void TDni_TextChanged(object sender, EventArgs e)
        {

        }

        private void BGuardar_Click(object sender, EventArgs e)
        {
            // Verificar si alguno de los TextBox está vacío
            if (string.IsNullOrWhiteSpace(TDni.Text) ||
                string.IsNullOrWhiteSpace(TNombre.Text) ||
                string.IsNullOrWhiteSpace(TApellido.Text))
            {
                // Mostrar un mensaje de error si algún campo está incompleto
                MessageBox.Show("Debe Completar todos los campos", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }
            else
            {
                // Crear la variable ask y mostrar un cuadro de diálogo de consulta
                DialogResult ask = MessageBox.Show("¿Seguro que desea insertar un nuevo Cliente?",
                                                   "Confirmar Inserción",
                                                   MessageBoxButtons.YesNo,
                                                   MessageBoxIcon.Question,
                                                   MessageBoxDefaultButton.Button1);


                if (ask == DialogResult.Yes)
                {

                    string clienteNombre = $"{TNombre.Text} {TApellido.Text}";
                    MessageBox.Show($"El Cliente: {clienteNombre} se insertó correctamente",
                                    "Guardar",
                                    MessageBoxButtons.OK,
                                    MessageBoxIcon.Information);

                    LModificar.Text = clienteNombre;
                }
                else
                {
                    // Opcional: manejar la acción cuando se selecciona "No"
                    MessageBox.Show("La inserción ha sido cancelada", "Cancelación", MessageBoxButtons.OK, MessageBoxIcon.Information);
                }
            }
        }
        private void BEliminar_Click(object sender, EventArgs e)
        {

            DialogResult ask = MessageBox.Show($"Está apunto de eliminar el Cliente: {TNombre.Text} {TApellido.Text}",
                                               "Confirmar Eliminación",
                                               MessageBoxButtons.YesNo,
                                               MessageBoxIcon.Exclamation,
                                               MessageBoxDefaultButton.Button2);

            // Evaluar la respuesta
            if (ask == DialogResult.Yes)
            {
                // Limpiar todos los campos y el label LModificar
                TDni.Clear();
                TNombre.Clear();
                TApellido.Clear();
                LModificar.Text = string.Empty;

                // Mostrar mensaje de confirmación de eliminación
                MessageBox.Show($"El Cliente: {TNombre.Text} {TApellido.Text} se eliminó correctamente",
                                "Eliminar",
                                MessageBoxButtons.OK,
                                MessageBoxIcon.Information);
            }
            else
            {
                // Mostrar mensaje de cancelación (opcional)
                MessageBox.Show("La eliminación ha sido cancelada", "Cancelación", MessageBoxButtons.OK, MessageBoxIcon.Information);
            }
        }

        private void panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void RBVaron_CheckedChanged(object sender, EventArgs e)
        {
            if (RBVaron.Checked)
            {
                // Cambia la imagen del PictureBox a la del varón
                pictureBox1.Image = Image.FromFile("../../../Images/male.jpeg");
            }
        }

        private void RBMujer_CheckedChanged(object sender, EventArgs e)
        {
            if (RBMujer.Checked)
            {
                // Cambia la imagen del PictureBox a la de la mujer
            }
        }

        private void checkBox1_CheckedChanged(object sender, EventArgs e)
        {

        }

        private void BSalir_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void pictureBox1_Click(object sender, EventArgs e)
        {

        }

        private void RBMujer_CheckedChanged_1(object sender, EventArgs e)
        {
            pictureBox1.Image = Image.FromFile("../../../Images/female.jpeg");

        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }

}
